sympy/core/expr.py

```diff
@@ -491,6 +496,18 @@ def is_constant(self, *wrt, **flags):
         True
         """
 
+        def check_denominator_zeros(expression):
+
+            from ..solvers import solveset
+            from ..solvers.solvers import denoms
+
+            free_symbols = expression.free_symbols
+
+            for symb in free_symbols:
+                if any(solveset(d, symb) is not S.EmptySet for d in denoms(expression)):
```

Comment:
I don't really like the idea of checking for solutions since you could have a denominator that is not zero but for which a solution could not be found. How about checking if all denominators are not zero (den.is_zero will be True, False or None for den in denoms(expr))?